app/src/main/java/nl/quantifiedstudent/watch/binary/VarInt.kt

```diff
@@ -0,0 +1,285 @@
+package nl.quantifiedstudent.watch.binary
+
+import nl.quantifiedstudent.watch.extensions.shl
+import java.io.IOException
+import java.io.InputStream
+import java.io.OutputStream
+import java.nio.ByteBuffer
+
+
+object VarInt {
+    /**
+     * Maximum encoded size of 32-bit positive integers (in bytes)
+     */
+    const val MAX_VARINT_SIZE = 5
+
+    /**
+     * maximum encoded size of 64-bit longs, and negative 32-bit ints (in bytes)
+     */
+    const val MAX_VARLONG_SIZE = 10
+
+    /** Returns the encoding size in bytes of its input value.
+     * @param i the integer to be measured
+     * @return the encoding size in bytes of its input value
+     */
+    fun varIntSize(i: Int): Int {
+        var i = i
+        var result = 0
+        do {
+            result++
+            i = i ushr 7
+        } while (i != 0)
+        return result
+    }
+
+    /**
+     * Reads a varint from src, places its values into the first element of dst and returns the offset
+     * in to src of the first byte after the varint.
+     *
+     * @param src source buffer to retrieve from
+     * @param offset offset within src
+     * @param dst the resulting int value
+     * @return the updated offset after reading the varint
+     */
+    fun getVarInt(src: ByteArray, offset: Int, dst: IntArray): Int {
+        var offset = offset
+        var result = 0
+        var shift = 0
+        var b: Int
+        do {
+            if (shift >= 32) {
+                // Out of range
+                throw IndexOutOfBoundsException("varint too long")
+            }
+            // Get 7 bits from next byte
+            b = src[offset++].toInt()
+            result = result or (b and 0x7F shl shift)
+            shift += 7
+        } while (b and 0x80 != 0)
+        dst[0] = result
+        return offset
+    }
+
+    /**
+     * Reads a varint from the current position of the given ByteBuffer and returns the decoded value
+     * as 32 bit integer.
+     *
+     *
+     * The position of the buffer is advanced to the first byte after the decoded varint.
+     *
+     * @param src the ByteBuffer to get the var int from
+     * @return The integer value of the decoded varint
+     */
+    fun getVarInt(src: ByteBuffer): Int {
+        var tmp: Int
+        if (src.get().also { tmp = it.toInt() } >= 0) {
+            return tmp
+        }
+        var result = tmp and 0x7f
+        if (src.get().also { tmp = it.toInt() } >= 0) {
```

Comment:
Perhaps you could refactor this a bit. Only the number next to shl changes in each if-statement. There might be a more elegant solution.